src/D2L.CodeStyle.Analyzers/Immutability/ImmutableDefinitionChecker.cs

```diff
@@ -432,7 +459,7 @@ private enum AssignmentQueryKind {
 					// Some methods are known to have return values that are
 					// immutable (such as Enumerable.Empty()).
 					// These should be considered immutable by the Analyzer.
-					semanticModel ??= m_compilation.GetSemanticModel( assignment.Expression.SyntaxTree );
+					SemanticModel semanticModel = m_compilation.GetSemanticModel( assignment.Expression.SyntaxTree );

```

Comment:
I wonder if the symbol info should be part of `AssignmentInfo`, as we are passing in the same semantic model to the `Create`

Reply:
Yeah - somehow not calling `GetSemanticModel` everywhere would be good - they cache information for the syntax tree internally. At the same time, don't want to hold onto them longer than necessary, but that's probably the same lifetime as the syntax node itself anyway (as opposed to a syntax reference).